exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSystemPlugin.java

```diff
@@ -104,6 +108,31 @@ public FileSystemPlugin(FileSystemConfig config, DrillbitContext context, String
     }
   }
 
+  private boolean isS3() {
+    java.net.URI uri = FileSystem.getDefaultUri(fsConf);
+    return uri.getScheme().equals("s3a");
+  }
+
+  /**
+   * Retrieve secret and access keys from configured (with
+   * {@link org.apache.hadoop.security.alias.CredentialProviderFactory#CREDENTIAL_PROVIDER_PATH} property)
+   * credential providers and set it into {@link #fsConf}. If provider path is not configured or credential
+   * is absent in providers, it will conditionally fallback to configuration setting. The fallback will occur unless
+   * {@link org.apache.hadoop.security.alias.CredentialProvider#CLEAR_TEXT_FALLBACK} is set to false.
+   * @throws IOException thrown if a credential cannot be retrieved from provider
+   */
+  private void handleS3Credentials() throws IOException {
+    final String[] credentialKeys = {"fs.s3a.secret.key", "fs.s3a.access.key"};
+    for (String key : credentialKeys) {
+      char[] credentialChars = fsConf.getPassword(key);
+      if (credentialChars != null) {
+        fsConf.set(key, String.valueOf(credentialChars));
```

Comment:
s3a connector actually calls trim() on the created string, so that you can have some whitespace around any inline declaration. Worth retaining for consistency

Reply:
Hm, what do you call an "inline declaration"? In s3 storage plugin and with hadoop credential CLI's -value (though this option is for testing because it's unsecure) the value should be wrapped with quotes if it is designed to contain spaces. If set in core-site.xml, the value is wrapped with tags, so any space is considered to be intentional.